### ISSUE TITLE
Users with the Adult role in one family may be added to the Child role in another

### ISSUE THREAD

**Issue by mooreds** [FusionAuth/fusionauth-issues#737]
### Description 

Users with the 'adult' family role are only allowed to be in one family. However, you can add a user who has the adult role in a family to another family as a child.

### Affects versions
1.17.4

### Steps to reproduce
* create a user with a birthday of 1980-01-01
* create a family with that user
* create a second user with a birthday of 1980-01-01
* create a family with that user
* now add the first user as an child and to the second family you just created
* you are allowed to do so.

### Expected behavior

A user should not be able to be added to a second family as a child if they are part of any other family as an adult.

### Platform

N/A

### Additional context

Found as part of the family guide preparation.

**Comment by robotdan** [comment]
@voidmain do you remember if this was intentional? I know we wanted to keep it flexible. IRL - you could be an adult with a child, and still be a child of someone else.

**Comment by mooreds** [comment]
Fair point. I was fixated on child being defined by age when I filed, but perhaps that's overly narrow.

**Comment by mooreds** [comment]
Whoopsie, hit the wrong button.

**Comment by voidmain** [comment]
I don't know if we defined it that clearly. The family roles might need some work honestly. The `adult`, `teen` and `child` seem to indicate age in some ways. The `owner` does not indicate age as much. Perhaps the `ageDefinition` should be dynamic based on ages but the `roles` should be defined.
